src/main/java/com/endava/endabank/dto/user/UserRegisterDto.java

```diff
@@ -1,17 +1,50 @@
 package com.endava.endabank.dto.user;
 
 
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.*;
 
 @Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
 public class UserRegisterDto {
     private Integer id;
+    @Email
+    @NotNull
     private String email;
+
+    @NotNull
+    @Size(min = 10,max = 20)
     private String phoneNumber;
+
+    @NotNull
+    @Size(min = 6,max = 20)
     private String identifier;
+
+    @NotNull
+    @NotBlank
     private String firstName;
+
+    @NotNull
+    @NotBlank
     private String lastName;
+
+    @NotNull
+    @Size(min = 8, max = 20)
+    @Pattern(regexp = "((?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[\\W]).{8,20})",
+            message = "1 Capital, 1 Special character, 1 Number and 8 to 20 digits,")
     private String password;
-    private Integer roleId;
+
+    @NotNull
+    @Digits(fraction = 0,integer = 2, message = "Number between 1 and 99")
     private Integer typeIdentifierId;
+
+    public void setEmail(String email){
+        this.email = email.toLowerCase();
```

Comment:
Hi,

When you have a dto class is not a good practice to have logic like this toLowerCase(), it would be better if you have that part in another utility class or when you need it.

http://www.juntadeandalucia.es/servicios/madeja/sites/default/files/historico/1.3.0/contenido-libro-pautas-38.html

"Un DTO normalmente no provee lógica de negocios o validaciones de ningún tipo. Solo provee acceso a las propiedades del bean."